lib/protobuf/nats/client.rb

```diff
@@ -1,10 +1,30 @@
+require "connection_pool"
 require "protobuf/nats"
 require "protobuf/rpc/connectors/base"
 require "monitor"
 
 module Protobuf
   module Nats
     class Client < ::Protobuf::Rpc::Connectors::Base
+      # Structure to hold subscription and inbox to use within pool
+      SubscriptionInbox = ::Struct.new(:subscription, :inbox)
+
+      def self.subscription_pool
+        @subscription_pool ||= ::ConnectionPool.new(:size => subscription_pool_size, :timeout => 0.1) do
```

Comment:
I wonder if we should allow this to dynamically grow since we're just trying to avoid mutating the subscriptions path too much.